### DISCUSSION TITLE
Xcode + `index-import` from CI

### DISCUSSION THREAD

**Discussion by mattrobmattrob** [MobileNativeFoundation/discussions#190] [General]
How are folks using [`index-import`](https://github.com/MobileNativeFoundation/index-import) to import prebuilt indexes from CI?

There's precedent in [`rules_ios`](https://github.com/bazel-ios/rules_ios) to call it within a post build script when Bazel is used as the build system as a proxy for the index-while-building (`COMPILER_INDEX_STORE_ENABLE`) functionality in Xcode/`xcodebuild`. I'm less interested in that since it's fairly understandable/defined.

But...for those that are importing it "before" any build/before opening Xcode for engineers:

- When are you doing that import? As part of project generation? Pre build script?
- How do you mitigate any potential slowness if the import is large (similar to https://github.com/MobileNativeFoundation/discussions/discussions/60)?
- Do you try to do a large build to generate a maximal index or focus on a single target like your main app?

**Comment by 4itdg3x9** [comment]
Sharding: If you have 500+ targets, don't import the full index store. Import the index for the Main App target and its direct feature-layer dependencies. This provides the most "bang for buck" for autocomplete and jump-to-definition.
Differential Imports: Don't replace the local index entirely if you can [N1BET Casino](https://n1betcasino-au.com/) avoid it. Use a smaller "base" index that covers your most frequently touched internal libraries, and let Xcode's native indexbuild handle the edge cases.
Asynchronous Sync: Some teams use a hidden helper script that fetches the artifact from S3/GCS while the engineer is grabbing coffee after the initial repo sync.
